File: Domain/Consultation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class Consultation
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string PatientName { get; set; }

        public DateTime PatientBirthDate { get; set; }

        public string PatientEmail { get; set; }

        public string PatientPhone { get; set; }

        [ForeignKey("ProblemId")]
        public Problem Problem { get; set; }

        public int ProblemId { get; set; }

        public Psychologist Psychologist { get; set; }

        public bool IsRemote { get; set; }

        public string Address { get; set; }

        public int Date { get; set; }

        public int Duration { get; set; }

        public int Bonus { get; set; }

        public int Cost { get; set; }

    }
}

File: bettercalm-frontend/src/app/models/consultation.ts
import { Injectable } from '@angular/core';
import { Adapter } from './adapter';

export class Consultation {
    constructor(
        public id: number,
        public patientName: string,
        public patientBirthDate: Date,
        public patientEmail: string,
        public patientPhone: string,
        public problemId: number,
        public psychologistId: number,
        public isRemote: boolean,
        public address: string,
        public date: number,
        public duration: number,
        public bonus: number
    ) { }
}

@Injectable({
    providedIn: 'root'
})
export class ConsultationAdapter implements Adapter<Consultation> {
    adapt(item: any): Consultation {
        return new Consultation(
            item.id,
            item.patientName,
            item.patientBirthDate,
            item.patientEmail,
            item.patientPhone,
            item.problemId,
            item.psychologistId,
            item.isRemote,
            item.address,
            item.date,
            item.duration,
            item.bonus
        )
    }
}